src/main/java/ch/wisv/areafiftylan/web/tournament/service/TournamentServiceImpl.java

```diff
@@ -30,6 +31,17 @@ public Tournament addTournament(Tournament tournament) {
         return tournamentRepository.save(tournament);
     }
 
+    @Override
+    public Tournament replaceTournament(Tournament tournament) {
+        if (getTournaments()
```

Comment:
So you're iterating over all tournaments, checking if none match the given ID? Why not just use [this](https://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html#getOne-ID-)